kratos/includes/registry_value_item.h

```diff
@@ -0,0 +1,226 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Pooyan Dadvand
+//
+
+#if !defined(KRATOS_REGISTRY_VALUE_ITEM_H_INCLUDED )
+#define  KRATOS_REGISTRY_VALUE_ITEM_H_INCLUDED
+
+
+// System includes
+#include <string>
+#include <iostream>
+
+
+
+// External includes
+
+
+// Project includes
+#include "includes/registry_item.h"
+
+
+namespace Kratos
+{
+///@addtogroup ApplicationNameApplication
+///@{
+
+///@name Kratos Globals
+///@{
+
+///@}
+///@name Type Definitions
+///@{
+
+///@}
+///@name  Enum's
+///@{
+
+///@}
+///@name  Functions
+///@{
+
+///@}
+///@name Kratos Classes
+///@{
+
+/// The registry value item is a registry item which stores a copy of a value.
+/** This class is derived from RegistryItem and the difference is that it creates a 
+ *  copy of the value in initialization and delete that copy at destruction. 
+ *  This make it suitable for storing values that we want to have one copy of them 
+ *  in the registry but not for the values that are stored in other places (like static variables) 
+ *  Value type should be copy constructable.
+*/
+template<typename TValueType>
+class KRATOS_API(KRATOS_CORE) RegistryValueItem : public RegistryItem
```

Comment:
This `KRATOS_API` is causing the win problems. It should not be here unless there is a cpp for RegistryValueItem.

Reply:
You are right! Thanks for the hint!